x-pack/plugins/fleet/server/services/preconfiguration.ts

```diff
@@ -172,19 +178,57 @@ export async function ensurePreconfiguredPackagesAndPolicies(
     );
   }
 
+  // Install bundled packages first before we install any preconfigured packages
+  const bundledInstallResults = await installBundledPackages(soClient, esClient, spaceId);
+
+  const packagesToInstall = [];
+
+  for (const pkg of packages) {
+    // We don't support preconfiguring bundled packages. Preconfigured packages will be installed from the package
+    // registry if it is available while bundled packages will be installed from disk. In order to avoid version conflicts
+    // we only support one or the other, preferring bundled packages if they are avaiable.
+    if (bundledInstallResults.some((result) => result.name === pkg.name)) {
+      logger.warn(
+        `Preconfigured package ${pkg.name} will be skipped as a bundled version of this package is included with Kibana`
+      );
+
+      continue;
+    }
+
+    if (pkg.version === PRECONFIGURATION_LATEST_KEYWORD) {
+      packagesToInstall.push(pkg.name);
+    } else {
+      packagesToInstall.push(pkg);
+    }
+  }
+
   // Preinstall packages specified in Kibana config
   const preconfiguredPackages = await bulkInstallPackages({
     savedObjectsClient: soClient,
     esClient,
-    packagesToInstall: packages.map((pkg) =>
-      pkg.version === PRECONFIGURATION_LATEST_KEYWORD ? pkg.name : pkg
-    ),
+    packagesToInstall,
     force: true, // Always force outdated packages to be installed if a later version isn't installed
     spaceId,
   });
 
   const fulfilledPackages = [];
   const rejectedPackages: PreconfigurationError[] = [];
+
+  for (const bundledInstallResult of bundledInstallResults) {
+    if (bundledInstallResult.error) {
+      logger.warn(
+        `Failed to install bundled package ${bundledInstallResult.name} due to error [${bundledInstallResult.error}]`
+      );
+
+      rejectedPackages.push({
+        package: { name: bundledInstallResult.name, version: '' },
```

Comment:
Would it be useful if `installBundledPackages` returned the package version as well? I guess if there were multiple versions of the same package bundled we may not know which one failed?